tuning/autotune.py

```diff
@@ -57,6 +58,88 @@ class CandidateTracker:
     calibrated_benchmark_diff: Optional[float] = None
 
 
+@dataclass(frozen=True)
+class PathConfig:
+    # Preset constants
+    global_config_prolog_mlir: Path = Path("./config_prolog.mlir")
+    global_config_epilog_mlir: Path = Path("./config_epilog.mlir")
+    compile_candidate_sh: Path = Path("./compile_candidate.sh")
+    benchmark_dispatch_sh: Path = Path("./benchmark_dispatch.sh")
+    compile_unet_candidate_sh: Path = Path("./compile_unet_candidate.sh")
+    benchmark_unet_candidate_sh: Path = Path("./benchmark_unet_candidate.sh")
+    unet_baseline_vmfb: Path = Path("./unet_baseline.vmfb")
+
+    # Dynamic paths
+    base_dir: Path = field(init=False)
+    local_config_prolog_mlir: Path = field(init=False)
+    local_config_epilog_mlir: Path = field(init=False)
+    template_mlir: Path = field(init=False)
+    candidates_dir: Path = field(init=False)
+    candidate_configs_pkl: Path = field(init=False)
+    compiled_dir: Path = field(init=False)
+    compilefailed_dir: Path = field(init=False)
+    candidate_vmfbs_txt: Path = field(init=False)
+    dispatch_benchmark_result_log: Path = field(init=False)
+    dispatch_benchmark_top_result_log: Path = field(init=False)
+
+    unet_benchmark_result_log: Path = field(init=False)
+    unet_result_log: Path = field(init=False)
+    candidate_trackers_pkl: Path = field(init=False)
+
+    # To be set outside of class
+    log_file_path: Optional[Path] = field(init=False, default=None)
+
+    def __post_init__(self):
+        object.__setattr__(self, "base_dir", self._create_base_dir())
```

Comment:
I think this shouldn't create any directories because it will make it difficult to test. It'd be better to run the code that initializes paths after this object's been constructed